proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1984 @@
+
+  * Let `new-alpha` be the result of [percent-converting] `$alpha` with a max
+    of 1.
+
+  * Set `alpha` to the value of `new-alpha + alpha` clamped between 0 and 1.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `key` and value `adjust` in `channel-args`:
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.adjust($color, $channel1: 0.25)` for adjusting
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set `adjust` to the result of [validating] `adjust` as `key` channel in
+    `space`.
+
+    > This ensures that `adjust` is percent-converted when necessary, and
+    > throws errors when the adjustment is not a valid value for the space.
+
+  * Set the `channel` with name or index of `key` in `channels` to
+    `channel + adjust`, treating values of `none` as `0`.
+
+* Let `new` be a color in color space `space`, with `channels` channels, and an
+  alpha of `alpha`.
+
+* Return the result of [converting] `new` into `origin-space`.
+
+### `color.scale()`
+
+```
+scale($color, $args...)
+```
+
+This function is also available as a global function named `scale-color()`.
+
+* If `$color` is not a color in a [known color space], throw an error.
+
+* If any item in `$args` is not a keyword argument, throw an error.
+
+* Let `origin-space` be `$color`'s color space.
+
+* If the keyword argument `$space` is specified in `$args`:
+
+    * Let `space` be the result of [looking up a known color space] named
+      `$space`.
+
+    * If `space` is not a [known color space], throw an error.
+
+      > We cannot scale channels in an unknown space, since there are no
+      > defined boundaries for a given channel to scale in relation to.
+
+    * Let `color` be the result of [converting] `$color` to `space`.
+
+* Otherwise:
+
+  * Let `space` be `origin-space`.
+
+  * Let `color` be the value of `$color`.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+    * If `alpha == none`, throw an error.
+
+    * Set `alpha` to the result of [scaling] `alpha` by `$alpha` with `max` 1.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `scale`, `factor` in `channel-args`:
+
+  * If `scale` is not the name of a [scalable] channel in `channels`, throw an
+    error.
+
+  * Let `channel` be the corresponding `channel` in `channels` with a name
+    matching `scale`.
+
+  * Let `max` be the upper boundary of `channel` in `space`.
+
+  * If `channel > max` or `channel < 0`, throw an error.
```

Comment:
I'd like to avoid throwing errors in these functions as much as possible unless the user is doing something clearly incoherent. We can reasonable scale sub-0 channels up and supramaximum channels down; if the user tries to scale them in the opposite direction, I think we should just keep them as-is, since that's the asymptotic behavior approaching min/max.